Formula/p/purr.rb

```diff
@@ -0,0 +1,35 @@
+class Purr < Formula
+  desc "Versatile zsh CLI tool for viewing and searching through Android logcat output"
+  homepage "https://github.com/google/purr"
+  url "https://github.com/google/purr/archive/refs/tags/2.0.4.tar.gz"
+  sha256 "ce8b4d31d6b56e79808f12a37795ea15127f3e01eb94f2becb1ee1cd8724844a"
+  license "Apache-2.0"
+  head "https://github.com/google/purr.git", branch: "main"
+
+  depends_on "fzf"
+  uses_from_macos "zsh"
+
+  def install
+    system "make"
+    bin.install "out/purr"
+
+    # This is needed for test
+    system "make", "adb_mock", "file_tester", "OUTDIR=#{pkgshare}"
+    chmod 0755, "#{pkgshare}/adb_mock"
+    chmod 0755, "#{pkgshare}/file_tester"
+  end
+
+  def caveats
+    <<~EOS
+      At runtime, adb must be accessible from your PATH.
+
+      You can install adb from Homebrew Cask:
+        brew install --cask android-platform-tools
+    EOS
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/purr -v")
+    shell_output("#{pkgshare}/file_tester -a #{pkgshare}/adb_mock -p #{bin}/purr")
```

Comment:
The command that `shell_output` is running is returning a nonzero exit code. Is this expected? If so, you can denote the expected exit code as a second argument to `shell_output`, e.g. `shell_output("#{pkgshare}/file_tester -a #{pkgshare}/adb_mock -p #{bin}/purr", 1)`.

Also, `shell_output` only captures the output, it doesn't do anything with it. You can `assert_match` it to some expected stdout output like is done in the preceding line with the version check.

If you don't care about the command line output (if a non-zero exit code would be thrown on error so no need to check/assert on output), you can also use the `system pkgshare/"program", "arg1", "arg2", "arg3"` form to just execute it without output capture. Note that this will fail if the exit code is nonzero though (there's no way to set the expected exit code of a `system` call, it's assumed to be 0).

Reply:
Also, separately, please make sure the version in the commit message matches the version being shipped in the formula (I know it's easy to overlook).

Reply:
> The command that shell_output is running is returning a nonzero exit code on a few CI machines. Is this expected?

Nope, this is not expected. Looks like a bug

Reply:
> make sure the version in the commit message matches the version being shipped in the formula

fixed, thank you